R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+                       "pmp[greater]" = tableResults[["PMP_1"]],
+                       "type[less]" = gettext("H2: Smaller"),
+                       "BF[less]" = tableResults[["BF_02"]],
+                       "pmp[less]" = tableResults[["PMP_2"]])
+          }
+        }
+      } else {
+        if (options$hypothesis == "equalBiggerSmaller") {
+          row <- list(Variable=currentPair, "type[equal]" = ".", "BF[equal]"= ".", "pmp[equal]" = ".",
+                      "type[greater]"= ".", "BF[greater]" = ".", "pmp[greater]" = ".",
+                      "type[less]" = ".", "BF[less]" = ".", "pmp[less]" = ".")
+        } else {
+          row <- list(Variable=currentPair, "hypothesis[type1]" = ".", "BF[type1]"= ".", "pmp[type1]" = ".",
+                      "hypothesis[type2]" = ".", "BF[type2]" = ".", "pmp[type2]" = ".")
+        }
+      }
+      table$addRows(row, rowNames = currentPair)
+      
+      if(pair[[1]] == pair[[2]]){
+        table$addFootnote(message=gettext("Results not computed: The variables in this pair are the same."), colNames="Variable", rowNames=currentPair)
+      }
+      if(pair[[1]] == "" || pair[[2]] == ""){
+        table$addFootnote(message=gettext("Results not computed: The pair is incomplete."), colNames="Variable", rowNames=currentPair)
+      }
+      progressbarTick()
+    }
+  } else if (type %in% c("anova", "ancova", "regression", "sem")){
+    variables <- switch(type,
+                        "anova" = c(options[["dependent"]], options[["fixedFactors"]]),
+                        "ancova" = c(options[["dependent"]], options[["fixedFactors"]], unlist(options[["covariates"]])),
+                        "regression" = c(options[["dependent"]], unlist(options[["covariates"]])),
+                        "sem" = .bainSemGetUsedVars(.bainSemTranslateModel(options[["syntax"]], dataset), colnames(dataset)))
+    if (any(variables %in% missing)){
+      i <- which(variables %in% missing)
+      if (length(i) > 1){
+        if(type == "regression" || type == "ancova" || type == "sem"){
+          table$addFootnote(message = gettextf("The variables %1$s contain missing values, the rows containing these values are removed in the analysis.", paste(variables[i], collapse = ", ")), symbol=gettext("<b>Warning.</b>"))
+        } else if(type == "anova"){
+          table$addFootnote(message= gettextf("The variables %1$s and %2$s contain missing values, the rows containing these values are removed in the analysis.", variables[1], variables[2]), symbol=gettext("<b>Warning.</b>"))
+        }
+      } else if (length(i) == 1){
+        table$addFootnote(message = gettextf("The variable %1$s contains missing values, the rows containing these values are removed in the analysis.", variables[i]), symbol=gettext("<b>Warning.</b>"))
+      }
+    }
+    bainResult <- .bainGetGeneralTestResults(dataset, options, bainContainer, ready, type)
+    for (i in 1:(length(bainResult[["fit"]]$BF)-1)){
```

Comment:
`bainResult` might be `NULL`, right? Maybe I'm missing where you check the container